src/ScientificTypesBase.jl

```diff
@@ -15,10 +15,7 @@ export Scientific, Found, Unknown, Known, Finite, Infinite,
     Annotated, AnnotationOf, Multiset, Iterator,
     Compositional
 
-export scitype, scitype_union, elscitype, nonmissing, trait
```

Comment:
I think you may have thrown the baby out with the bath water here 🛀🏾 . 

I think it's okay to export `elscitype`, as we did before, and you will need this to get your tests passing again.

I do agree `scitype_union` should *not* be exported and we should explicitly label the method as private in any doc-string we have for it.